patches/server/0577-Add-support-for-hex-color-codes-in-console.patch

```diff
@@ -243,7 +243,12 @@ index 0000000000000000000000000000000000000000..b4d0b7ecd56ab952319946854168c129
 +        while (matcher.find()) {
 +            int format = LOOKUP.indexOf(Character.toLowerCase(matcher.group().charAt(1)));
 +            if (format != -1) {
-+                matcher.appendReplacement(buffer, ansi ? ansiCodes[format] : "");
++                String replacement = "";
++                if (ansi && !(format >= 16 && format != 21)) { // handle the upstream quirk at CraftChatMessage L115
```

Comment:
Generally, this might just be me, but this logic seems rather convoluted for what it is doing.
Especially in regards to the ansi check done again below.
Like, unless I am missing something, this logic boils down to

```java
String replacement = "";
if (ansi) {
    replacement = (format < 16 || format == 21) ? ANSI_RESET + ansiCodes[format] : ansiCodes[format];
}
matcher.appendReplacement(buffer, replacement);
``` 

This way we
a) get rid of two checks of the ansi bool for no reason
b) de morgans the negation into the expression, which I believe to arguably makes this a lot easier to understand.

Reply:
No, i disagree
The two check are intended, if paper wants to remove the quirk into the bukkit parser
The negation is also intended to follow the bukkit flow (if else), and since the format is already hard to understand (bc the format is an index of a static string to represent color char), i don't think it's worth enough. But in general if people wants to understand more they can still go to the linked line